main/src/ca/uwaterloo/flix/language/phase/Lexer.scala

```diff
@@ -594,13 +594,44 @@ object Lexer {
     matches
   }
 
+  /**
+    * Checks whether the following substring matches a keyword.
+    * Will advance the current position past the keyword if there is a match.
+    */
+  private def isMatchCurrent(keyword: String)(implicit s: State): Boolean = {
+    // Check if the keyword can appear before eof.
+    if (s.current.offset + keyword.length > s.src.data.length) {
+      return false
+    }
+
+    // Check if the next n characters in source matches those of keyword one at a time.
+    val start = s.current.offset
+    var matches = true
+    var offset = 0
+    while (matches && offset < keyword.length) {
+      if (s.src.data(start + offset) != keyword(offset)) {
+        matches = false
+      } else {
+        offset += 1
+      }
+    }
+
+    if (matches) {
+      for (_ <- 0 until keyword.length) {
```

Comment:
I guess there's nothing like `advance(n)`?

Reply:
no - this code is also just copied from the other match, this the -1 adjusted